d2/src/test/java/com/linkedin/d2/xds/TestXdsClientImpl.java

```diff
@@ -443,6 +443,37 @@ public void testHandleD2URICollectionResponseWithRemoval()
     Assert.assertEquals(actualData.getURIMap(), D2_URI_MAP_UPDATE_WITH_DATA1.getURIMap());
   }
 
+  @Test
+  public void testWildCardResourceSubscription()
```

Comment:
Can you add a test to validate that calling `watchAllXdsResources` multiple times (even if it's the same subscriber) also replays the updates, just like for `watchXdsResource`? That way it's consistent

Reply:
After dicussion, it's ok for current unit test